x-pack/plugins/observability/public/components/custom_threshold/components/preview_chart/preview_chart.tsx

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState, useEffect } from 'react';
+import { EuiEmptyPrompt, useEuiTheme } from '@elastic/eui';
+import { FillStyle, OperationType } from '@kbn/lens-plugin/public';
+import { DataView } from '@kbn/data-views-plugin/common';
+import { FormattedMessage } from '@kbn/i18n-react';
+import useAsync from 'react-use/lib/useAsync';
+import {
+  LensAttributes,
+  LensAttributesBuilder,
+  XYChart,
+  XYDataLayer,
+  XYLayerOptions,
+  XYReferenceLinesLayer,
+} from '@kbn/lens-embeddable-utils';
+
+import { IErrorObject } from '@kbn/triggers-actions-ui-plugin/public';
+import { i18n } from '@kbn/i18n';
+import {
+  Aggregators,
+  Comparator,
+  CustomMetricAggTypes,
+} from '../../../../../common/custom_threshold_rule/types';
+import { useKibana } from '../../../../utils/kibana_react';
+import { MetricExpression } from '../../types';
+import { AggMap, PainlessTinyMathParser } from './painless_tinymath_parser';
+
+interface PreviewChartPros {
+  metricExpression: MetricExpression;
+  dataView?: DataView;
+  filterQuery?: string;
+  groupBy?: string | string[];
+  error?: IErrorObject;
+}
+
+function PreviewChart({
+  metricExpression,
+  dataView,
+  filterQuery,
+  groupBy,
+  error,
+}: PreviewChartPros) {
+  const {
+    services: { lens },
+  } = useKibana();
+  const { euiTheme } = useEuiTheme();
+  const { metrics, timeSize, timeUnit, threshold, comparator, equation } = metricExpression;
+  const [attributes, setAttributes] = useState<LensAttributes>();
+  const [aggMap, setAggMap] = useState<AggMap>();
+  const [formula, setFormula] = useState<string>('');
+  const [thresholdReferenceLine, setThresholdReferenceLine] = useState<XYReferenceLinesLayer[]>();
+  const [chartLoading, setChartLoading] = useState<boolean>(false);
+  const formulaAsync = useAsync(() => {
+    return lens.stateHelperApi();
+  }, [lens]);
+
+  // Handle Lens error
+  useEffect(() => {
+    // Lens does not expose or provide a way to check if there is an error in the chart, yet.
+    // To work around this, we check if the element with class 'lnsEmbeddedError' is found in the DOM.
```

Comment:
The embeddable provides the `onLoad(isLoading: boolean, adapters?: Partial<DefaultInspectorAdapters>, output$?: Observable<LensEmbeddableOutput> )` event, in theory, the `output$` observer, or even the `adapter` would give you the info that an error occurred. Did you try it?

Reply:
Yes, I tried it. In case of error, the key `hasError` no longer exists. Actually, the entire observer object is undefined. While the error message appears in the chart.

Reply:
i will suggest to extend the embeddable to provide this state. This too me seems like something that will work but probably not easy to maintain in the long term. 

Extending in embeddable means other users of embeddable will also benefit.

Reply:
@shahzad31, according to the Lens team, there is ongoing work to do that (exposing an error state/prop).